isbnbot/normalize_isbns.py

```diff
@@ -0,0 +1,152 @@
+        makedirs(log_dir, exist_ok=True)
+        log_file_datetime = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+        log_file = log_dir + '/%s_%s.log' % (job_name, log_file_datetime)
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setLevel(logging.DEBUG)
+        file_handler.setFormatter(log_formatter)
+        self.logger.addHandler(file_handler)
+
+    @staticmethod
+    def isbn_needs_normalization(isbn: str) -> bool:
+        """
+        Returns True if the given ISBN is valid and needs to be normalized (hyphens removed, letters capitalized, etc.)
+        Returns False otherwise
+        """
+        try:
+            normalized_isbn = isbnlib.get_canonical_isbn(isbn)  # get_canonical_isbn returns None if ISBN is invalid
+        except IndexError:  # get_canonical_isbn will raise this sometimes for an invalid ISBN
+            return False
+        else:
+            return normalized_isbn and normalized_isbn != isbn
+
+    def run(self, dump_filepath: str) -> None:
+        """
+        Performs ISBN normalization (removes hyphens and capitalizes letters)
+
+        dump_filepath -- path to *.txt.gz dump containing editions that need to be operated on
+        """
+        if self.dry_run:
+            self.logger.info('dry_run set to TRUE. Script will run, but no data will be modified.')
+
+        header = {'type': 0,
+                  'key': 1,
+                  'revision': 2,
+                  'last_modified': 3,
+                  'JSON': 4}
+        comment = 'normalize ISBN'
+        with gzip.open(dump_filepath, 'rb') as fin:
+            for row_num, row in enumerate(fin):
+                row = row.decode().split('\t')
+                _json = json.loads(row[header['JSON']])
+                if _json['type']['key'] != '/type/edition': continue
+
+                isbns_by_type = dict()
+                if 'isbn_10' in _json:
+                    isbns_by_type['isbn_10'] = _json.get('isbn_10', None)
+                if 'isbn_13' in _json:
+                    isbns_by_type['isbn_13'] = _json.get('isbn_13', None)
+                if not isbns_by_type: continue
+
+                skip_flag = True
+                for _, isbns in isbns_by_type.items():
+                    for isbn in isbns:
+                        if self.isbn_needs_normalization(isbn):
+                            skip_flag = False
+                            break
+                    if skip_flag: break
+                if skip_flag: continue
+
+                olid = _json['key'].split('/')[-1]
+                edition = self.ol.Edition.get(olid)
+                if edition.type['key'] != '/type/edition': continue
+
+                for isbn_type, isbns in isbns_by_type.items():
+                    normalized_isbns = list()
+                    isbns = getattr(edition, isbn_type, [])
+                    for isbn in isbns:
+                        if not self.isbn_needs_normalization(isbn): continue
```

Comment:
This should be something like
```python
if not self.isbn_needs_normalization(isbn):
    normalized_isbns.append(isbn)
```
or maybe do an else case cause that will be more elegant